src/error.rs

```diff
@@ -189,161 +197,133 @@ pub enum KafkaCode {
     UnsupportedVersion = 35,
 }
 
-impl From<io::Error> for Error {
-    fn from(err: io::Error) -> Error {
-        if err.kind() == io::ErrorKind::UnexpectedEof {
-            Error::UnexpectedEOF
-        } else {
-            Error::Io(err)
-        }
-    }
-}
-
-#[cfg(feature = "security")]
-impl From<SslError> for Error {
-    fn from(err: SslError) -> Error {
-        Error::Ssl(err)
-    }
-}
-
-#[cfg(feature = "security")]
-impl From<ErrorStack> for Error {
-    fn from(err: ErrorStack) -> Error {
-        Error::SslHandshake(err)
-    }
-}
-
 #[cfg(feature = "security")]
 impl<S> From<ssl::HandshakeError<S>> for Error {
     fn from(err: ssl::HandshakeError<S>) -> Error {
         match err {
             ssl::HandshakeError::SetupFailure(e) => From::from(e),
             ssl::HandshakeError::Failure(s) |
-            ssl::HandshakeError::Interrupted(s) => from_sslerror_ref(s.error()),
+            ssl::HandshakeError::Interrupted(s) => from_sslerror_ref(s.error()).into(),
         }
     }
 }
 
-#[cfg(feature = "snappy")]
-impl From<::snap::Error> for Error {
-    fn from(err: ::snap::Error) -> Error {
-        Error::InvalidSnappy(err)
-    }
-}
-
 impl Clone for Error {
```

Comment:
Hmm, is it possible to just derive `Clone` and get rid of the rest of the module?